File: hammer/core/datahub/source.py
from sqlalchemy import JSON, Column, String, UniqueConstraint

from ..base import Base


class Source(Base):
    __tablename__ = "source"

    name = Column(String(256), nullable=False, primary_key=True)
    version = Column(String(256), nullable=False, primary_key=True)
    table_name = Column(String(256), nullable=False)
    infra_type = Column(String(256), nullable=False)
    field_mapping = Column(JSON, default=dict)
    owner = Column(String(256))
    description = Column(String(256))
    config = Column(JSON, default=dict)  # 数据源的配置信息

    __table_args__ = (UniqueConstraint("name", "version"),)

    def __repr__(self):
        return f"<Source {self.name}:{self.version}:{self.table_name}>"

File: server/src/main/java/com/hammer/entity/Source.java
package com.hammer.entity;

import lombok.Data;
import java.util.Map;


@Data
public class Source {
    private String name;
    private String version;
    private String tableName;
    private String infraType;
    private Map<String, String> fieldMapping;
    private String owner;
    private String description;
    private Map<String, String> config;

    public Source(String name, String version, String tableName, String infraType, Map<String, String> fieldMapping,
                  String owner, String description, Map<String, String> config) {
        this.name = name;
        this.version = version;
        this.tableName = tableName;
        this.infraType = infraType;
        this.fieldMapping = fieldMapping;
        this.owner = owner;
        this.description = description;
        this.config = config;
    }

    @Override
    public String toString() {
        return String.format("<Source %s:%s:%s>", name, version, tableName);
    }
}
